database/award_query.py

```diff
@@ -60,3 +60,37 @@ def _query_async(self):
             Award.team_list == ndb.Key(Team, team_key),
             Award.event == ndb.Key(Event, event_key)).fetch_async()
         raise ndb.Return(awards)
+
+
+class TeamYearTypeAwardsQuery(DatabaseQuery):
+    CACHE_VERSION = 0
+    CACHE_KEY_FORMAT = 'team_year_type_awards_{}_{}_{}'  # (team_key, year, award_type_enum)
+    DICT_CONVERTER = AwardConverter
+
+    @ndb.tasklet
+    def _query_async(self):
+        team_key = self._query_args[0]
+        year = self._query_args[1]
+        award_type_enum = self._query_args[2]
+        awards = yield Award.query(
+            Award.team_list == ndb.Key(Team, team_key),
+            Award.year == year,
+            Award.award_type_enum == award_type_enum).fetch_async()
+        raise ndb.Return(awards)
+
+
+class TeamEventsTypeAwardsQuery(DatabaseQuery):
```

Comment:
Ditto on cache clearing

Reply:
Let's rename this to `TeamEventTypeAwardsQuery`